src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/Policies/CamelCasingPolicy.cs

```diff
@@ -25,23 +25,7 @@ public static string FromJson(string value)
                 args.value.AsSpan(1).CopyTo(chars.Slice(1));
             });
 #else
-            if (value.Length < 256)
-            {
-                Span<char> span = stackalloc char[value.Length];
-                span[0] = char.ToUpperInvariant(firstChar);
-                value.AsSpan(1).CopyTo(span.Slice(1));
-                unsafe
-                {
-                    fixed (char* ptr = span)
-                    {
-                        return new string(ptr, 0, span.Length);
-                    }
-                }
-            }
-            else
-            {
-                return char.ToUpperInvariant(firstChar) + value.Substring(1);
-            }
+            return char.ToUpperInvariant(value[0]) + value.Substring(1);
```

Comment:
use the cached `firstChar` value

Also, not worth using stackalloc to avoid extra allocs in the common case?

Reply:
Yes I will change to use firstChar. However I am going to hold off on micro-optimizations for netstandard like unsafe stackalloc for now at least here because the camel-casing name is cached.